File: java/controllers/model/GameMap.java
package controllers.model;

import rooms.options.GameMasterable;

public class GameMap implements Mappable {
    private final String filePath;
    private int mapHash;
    private GameMasterable startRoom;
    private String shortestPath;

    public GameMap(String filePath, int mapHash, GameMasterable startRoom) {
        this.filePath = filePath;
        this.mapHash = mapHash;
        this.startRoom = startRoom;
        this.shortestPath = "";
    }

    @Override
    public String getMapFilePath() {
        return filePath;
    }

    @Override
    public GameMasterable getStartRoom() {
        return startRoom;
    }

    @Override
    public String getShortestPath() {
        return shortestPath;
    }

    @Override
    public void setShortestPath(String path) {
        this.shortestPath = path;
    }

    @Override
    public int hashCode(){
        return mapHash;
    }
}

File: php/controllers/model/GameMap.php
<?php

namespace Controllers\Model;

use Rooms\Options\GameMasterable;

class GameMap implements Mappable {
    private $filePath;
    private $mapHash;
    private $startRoom;
    private $shortestPath;

    public function __construct($filePath, $mapHash, GameMasterable $startRoom) {
        $this->filePath = $filePath;
        $this->mapHash = $mapHash;
        $this->startRoom = $startRoom;
        $this->shortestPath = "";
    }

    public function getMapFilePath() {
        return $this->filePath;
    }

    public function getStartRoom() {
        return $this->startRoom;
    }

    public function getShortestPath() {
        return $this->shortestPath;
    }

    public function setShortestPath($path) {
        $this->shortestPath = $path;
    }

    public function hashCode() {
        return $this->mapHash;
    }
}
?>


